src/ac/write.c

```diff
@@ -428,11 +428,17 @@ int writeac(X509 *issuerc, STACK_OF(X509) *issuerstack, X509 *holder, EVP_PKEY *
     }
   }
 
-  alg1 = X509_ALGOR_dup((X509_ALGOR*)X509_get0_tbs_sigalg(issuerc));
   {
-    X509_ALGOR /*const*/* sig_alg;
+    const X509_ALGOR *sig_alg = X509_get0_tbs_sigalg(issuerc);
+    alg1 = X509_ALGOR_dup((X509_ALGOR*)sig_alg); // const_cast
```

Comment:
I would document a bit why you need to do the const cast here and in line 441 below.
It seems to be a broken API definition in OpenSSL for `X509_ALGOR_dup()`, right? AFAICT `X509_ALGOR_dup()`
is defined using `OPENSSL_sk_dup()` which in itself seems to have a `const` pointer as its only parameter. So I think it's save for both OpenSSL 1.1 and 3.0.X (haven't looked at the OpenSSL < 1.1 code but I presume it's fine too).

Reply:
I prefer to leave it as it is